### PR TITLE
Poc/seeding part1

### PR DESCRIPTION
**Pull request by esteban-gs**
## Dev Summary

- type-safe and dependency-injected solution for seeding data into DB

## Test Plan

repro steps:

1. from `server` run `npm run seed:drop` to drop all data in database and seed with sample data (only users for now)
2. You can expect to see a successful run.
3. You should see the sample users in your dev database

## Resources

- [Drop all data](https://github.com/nestjs/nest/issues/409#issuecomment-406803921)
- [NESTJS Lifecycle events](https://docs.nestjs.com/fundamentals/lifecycle-events#lifecycle-events)
- [Seeding Strategy](https://medium.com/the-crowdlinker-chronicle/seeding-databases-using-nestjs-cd6634e8efc5)
- [Standalone app creation (bootstrapping)](https://docs.nestjs.com/faq/serverless#using-standalone-application-feature)

### CODE REVIEW

**Comment by jd2rogers2** [comment]
looking pretty good so far!
i was also playing around with this last week and came up with something a little more lightweight (but may have less capabilities down the road 🤷 )
will let you decide which is best route to take
but here it is https://github.com/Nonprofit-Exchange-Hub/web-app/pull/146

**Comment by esteban-gs** [comment]
> looking pretty good so far! i was also playing around with this last week and came up with something a little more lightweight (but may have less capabilities down the road shrug ) will let you decide which is best route to take but here it is #146

I think I'll stay with this solution since, it'll be easier to maintain down the line with built-in services and type-safety.

File: server/src/database/seeding/seed.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { INestApplicationContext } from '@nestjs/common';
+import { NestFactory } from '@nestjs/core';
+import { SeederModule } from 'src/seeder/seeder.module';
```
Comment (jd2rogers2) [inline server/src/database/seeding/seed.ts:3]: relative path

File: server/src/database/seeding/dev-seed.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { CreateUserDto } from 'src/users/dto/create-user.dto';
```
Comment (jd2rogers2) [inline server/src/database/seeding/dev-seed.ts:1]: should be relative path
so `../../users/dto/create-user.dto` i think

File: server/src/seeder/seeder.service.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { users } from 'src/database/seeding/dev-seed';
+import { CreateUserDto } from 'src/users/dto/create-user.dto';
+import { UsersService } from 'src/users/users.service';
```
Comment (jd2rogers2) [inline server/src/seeder/seeder.service.ts:4]: check all your paths plz

File: server/src/database/seeding/dev-seed.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    {
+      id: 0,
+      first_name: 'user2First',
+      last_name: 'user2Last',
+      email: 'user2First@example.com',
+      password: 'Secret1234%',
+    },
+  ];
+  return users;
+};
```
Comment (jd2rogers2) [inline server/src/database/seeding/dev-seed.ts:21]: is the intent for this file to house all of the data? like beyond just user?
the filename doesn't clarify for me
maybe `seed-user`, `seed-data`, or `dev-data`
and below file could be `seed-runner` 🤷

File: server/src/seeder/seed-database-connection.service.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      port: Number(process.env.DATABASE_PORT),
+      username: process.env.DATABASE_USER,
+      password: process.env.DATABASE_PASSWORD,
+      database: process.env.DATABASE_DB,
+      synchronize: true, //shouldn't be used in production
+      dropSchema: false, //toggle to true to clear database schema
+      logging: true,
+      entities: ['./src/**/*.entity.ts'],
+    };
+  }
```
Comment (jd2rogers2) [inline server/src/seeder/seed-database-connection.service.ts:26]: could be separate PR but all the db connection configs seem to be getting redundant

File: server/package.json [hunk 6]
```diff
@@ -21,7 +21,8 @@
         "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
         "test:e2e": "jest --config ./test/jest-e2e.json",
         "checks": "npm run test:e2e && npm run test && npm run lint",
-        "prepare": "cd .. && husky install && cd server"
+        "prepare": "cd .. && husky install && cd server",
+        "seed:drop": "ts-node -r tsconfig-paths/register src/database/seeding/seed.ts"
```
Comment (jd2rogers2) [inline server/package.json:25]: this command naming to me says we're dropping the db/data
but it's really doing that and re-seeding
maybe `dropAndSeed`?

File: server/src/seeder/seed-database-connection.service.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@nestjs/common';
```
Comment (jd2rogers2) [inline server/src/seeder/seed-database-connection.service.ts:1]: why different folders? like why not put everything in one?

**Review by jd2rogers2 — COMMENTED** [review]
everything looks really good (apart from some small nits)
but it didn't work for me 🤔 
db still has old data and no new data, see screenshot showing that my .env values match the db i'm querying
Screen Shot 2022-03-22 at 6 33 35 PM

**Comment by esteban-gs** [comment]
I pushed fixes to your review. You should be able to run it now.

File: server/src/seeder/seeder.module.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
+import { Logger, Module, OnApplicationBootstrap } from '@nestjs/common';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { UsersModule } from '../users/users.module';
+import { SeederService } from './seeder.service';
+import { DatabaseConnectionService } from 'src/database-connection.service';
```
Comment (jd2rogers2) [inline server/src/seeder/seeder.module.ts:5]: one more path that should be relative

**Review by jd2rogers2 — APPROVED** [review]
🌱 🌱 🌱 🌱 🌱🌱  
🌱 success!!! 🌱 
🌱 🌱 🌱 🌱 🌱 🌱 

(one more path to fix real quick before merge though)

**Pull request merged**